tensorflow/python/estimator/estimator.py

```diff
@@ -207,7 +207,7 @@ def __init__(self, model_fn, model_dir=None, config=None, params=None,
     else:
       self._session_config = self._config.session_config
 
-    self._device_fn = _get_replica_device_setter(self._config)
+    # self._device_fn = _get_replica_device_setter(self._config)
```

Comment:
I think we might need to keep this line such that 
_get_replica_device_setter should respect config.device_fn first and, if missing, create a good default like the old behavior. 

wdyt?